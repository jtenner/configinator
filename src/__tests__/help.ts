import { help, parse, UserDefinedConfigration } from "../index";

const exampleConfig: UserDefinedConfigration = {
  config: {
    name: "config",
    type: "R",
    description: "A user passed configuration location",
  },

  test1: {
    name: "test1",
    type: "b",
    description: "A boolean flag",
  },
  nodesc: {
    name: "nodesc",
    type: "s",
  },
};

test("help text", () => {
  const result = parse([], exampleConfig, {
    cwd: process.cwd(),
    readFileSync(name: string, basename: string): string | null {
      return null;
    },
  });
  const helpText = help(result, {});
  expect(result.diagnostics).toMatchSnapshot("help text diagnostics");
  expect(helpText).toMatchSnapshot("help text");
});
