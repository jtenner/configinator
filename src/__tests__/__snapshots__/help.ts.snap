// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help text: help text 1`] = `
"      --config: [Node Require] A user passed configuration location
      --test1: [Boolean Flag] A boolean flag
      --nodesc: [String]"
`;

exports[`help text: help text diagnostics 1`] = `Array []`;
